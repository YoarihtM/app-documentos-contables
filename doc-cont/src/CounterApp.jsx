import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = (  ) => {

        // setCounter( counter + 1 );

        setCounter( ( c ) => c + 1 );
    };
    
    const handleSustract = (  ) => {

        // setCounter( counter + 1 );

        setCounter( ( c ) => c - 1 );
    };

    const handleReset = (  ) => {

        // setCounter( counter + 1 );

        setCounter( ( c ) => value );
    };

    return(
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ handleAdd }>
                +1
            </button>
            <button onClick={ handleSustract }>
                -1
            </button>
            <button onClick={ handleReset }>
                Reset
            </button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}