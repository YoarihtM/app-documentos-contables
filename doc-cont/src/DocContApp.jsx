import PropTypes from 'prop-types';

const titulo = 'Documentos Contables';
const funcPrueba = () => 'Valor';

export const DocContApp = ({ title, subtitle }) => {

    // console.log( props );

    // if( !title ) throw new Error('El "title" no existe');

    return (
        <>
            <h1> { title } </h1>
            <h4> { subtitle } </h4>
            <p> { funcPrueba() } </p>
            <div>hola</div>
        </>
    );
};
