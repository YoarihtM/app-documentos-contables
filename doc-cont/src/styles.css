html,
body {
    background-color: #21232a;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}