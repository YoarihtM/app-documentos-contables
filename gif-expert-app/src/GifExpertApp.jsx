import { useState } from "react"
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // cuando se requiere almacenar la información y esa información
    // tiene que cambiar el html se debe almacenar la información
    // se requiere un hook de react para mantener el estado

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = (newCategory) => {

        // console.log(newCategory)

        if(categories.includes(newCategory)) return;

        setCategories([ newCategory,...categories  ]);

        // setCategories( (c) => c.push('Valorant'));
        // categories.push('Valorant');
        // setCategories(categories)

        // Evitar usar push, por eso no se actualiza la vista
        // push muta el objeto y react evita mutar el objeto

        // setCategories( cat => [ ...cat, 'Valorant' ]);
        // setCategories(['Valorant', ...categories]);

    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                onNewCategory={ (value) => onAddCategory(value) }
                
            />

            {categories.map(category => 
                (
                    <GifGrid 
                        key={category} 
                        category={category}
                    />
                ))
            }
        </>
    )
}
