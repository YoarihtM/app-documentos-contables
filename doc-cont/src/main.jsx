import React from "react";
import ReactDOM from "react-dom/client";
import { DocContApp } from './DocContApp';
import './styles.css'
import { CounterApp } from "./CounterApp";


ReactDOM.createRoot(document.getElementById('root')).render(

    <React.StrictMode>
        {/* <DocContApp title="Hola, soy vegeta" /> */}
        <CounterApp />
    </React.StrictMode>
);