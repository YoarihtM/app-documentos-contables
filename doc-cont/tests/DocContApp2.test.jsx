import { render, screen } from '@testing-library/react';
import { DocContApp } from '../src/DocContApp';


describe('pruebas en DocContApp', () => {

    const title = 'Hola, Soy Goku';
    const subtitle = 'Soy Goku un subtitulo';

    test('debe hacer match con el snapshot', () => { 
        
        const { container } = render( <DocContApp title={ title } />);
        expect( container ).toMatchSnapshot();
        
    });

    test('debe de mostrar el mensaje "Hola, Soy Goku"', () => { 

        // screen es lo que se muestra en la pantalla
        // es decir el renderizado del componente
        render( <DocContApp title={ title } />);
        expect( screen.getByText( title ) ).toBeTruthy();

        // screen.debug();

    });

    test('debe de mostrar el titulo en un H1', () => { 

        render( <DocContApp title={ title } />);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );

    });

    test('debe de mostrar el subtitulo enviado por props', () => { 

        render( 
            <DocContApp 
                title = { title } 
                subtitle = { subtitle } 
            />
        );

        expect( screen.getAllByText(subtitle).length ).toBe(2);

    });

})