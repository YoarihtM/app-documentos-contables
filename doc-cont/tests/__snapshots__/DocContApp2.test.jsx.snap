// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en DocContApp debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <h4>
     
    No hay subtítulo
     
  </h4>
  <h4>
     
    No hay subtítulo
     
  </h4>
  <p>
     
    Yoariht Macedo
     
  </p>
</div>
`;
